apps/web/src/components/CrossChainVeCakeModal/hooks/useCrossChainMessage.ts

```diff
@@ -0,0 +1,18 @@
+import { ChainId } from '@pancakeswap/chains'
+import { CrossChainMessage, getCrossChainMessage } from '@pancakeswap/ifos'
+import { useQuery } from '@tanstack/react-query'
+import { FAST_INTERVAL } from 'config/constants'
+
+export type CrossChainStatus = CrossChainMessage['status']
+export const useCrossChianMessage = (targetChainId?: ChainId, txHash?: string) => {
+  const { data, isLoading } = useQuery({
+    queryKey: ['crossChainMessage', targetChainId, txHash],
+    queryFn: () => {
+      if (!txHash || !targetChainId) return undefined
```

Comment:
let's throw error instead of returning undefined if required condition doesn't match